lib/zip_dir/unzipper.rb

```diff
@@ -22,16 +22,38 @@ def unzipped?
     def unzip
       ::Zip::File.open(zip_path) do |zip_file|
         zip_file.each do |entry|
+          next if unauthorized_file?(entry)
+
           file_path = "#{@unzip_path}/#{entry.name}"
 
           # Fixes: spec/fixtures/encoded_differently.zip (in a test)
-          dir_path = File.dirname(file_path).to_s
-          FileUtils.mkdir_p dir_path unless File.exists?(dir_path)
+          # dir_path = File.dirname(file_path).to_s
+          # FileUtils.mkdir_p dir_path unless File.exists?(dir_path)
+          ensure_path_exists file_path
 
           entry.extract(file_path) unless File.exists?(file_path)
         end
       end
       @unzipped = true
     end
+
+    private
+
+    def unauthorized_file?(file)
+      file.name.match(/MACOSX|.DS_Store/)
+    end
+
+    def ensure_path_exists file_path
+      # Create directory structure if doesn't exists
+      if file_path.to_s.match(/(?<last_caracter>\/$)/)
```

Comment:
test please.... i have stronger requirements for gems. plus `last_caracter` is misspelled......